deposit/src/main/java/edu/unc/lib/deposit/CleanupDepositJob.java

```diff
@@ -103,35 +113,23 @@ public void runJob() {
 		getJobStatusFactory().expireKeys(getDepositUUID(),
 				this.getStatusKeysExpireSeconds());
 	}
-	
+
 	private void deleteStagedFiles(Model m) {
 		NodeIterator ni = m.listObjectsOfProperty(CdrDeposit.stagingLocation);
 		try {
 			while (ni.hasNext()) {
 				RDFNode n = ni.nextNode();
 				URI stagingUri = URI.create(n.asLiteral().getString());
-				
-				SharedStagingArea area = getStorageArea(stagingUri);
-				if (area == null) {
-					continue;
-				}
-				
-				URI storageUri = null;
-				try {
-					storageUri = area.getStorageURI(stagingUri);
-				} catch (StagingException e) {
-					LOG.error("Could not resolve storage URI: {}", stagingUri.toString(), e);
-				}
-				
-				CleanupPolicy p = area.getIngestCleanupPolicy();
-				switch (p) {
+
+				CleanupPolicy cleanupPolicy = stagingManager.getCleanupPolicy(stagingUri);
+				switch (cleanupPolicy) {
 				case DO_NOTHING:
 					break;
 				case DELETE_INGESTED_FILES:
-					deleteFile(storageUri);
+					deleteFile(stagingUri);
 					break;
 				case DELETE_INGESTED_FILES_EMPTY_FOLDERS:
-					File parent = deleteFile(storageUri);
+					File parent = deleteFile(stagingUri);
 					if (parent != null && parent.exists()) {
 						if (parent.list().length == 0) {
```

Comment:
Eclipse's indentation for switch blocks is kind of weird in my opinion, but i opted not to fight it.  Or are you mainly referring to the fact that the code gets pretty far indented?